src/python/pants/backend/jvm/tasks/nailgun_task.py

```diff
@@ -19,12 +19,19 @@
 
 class NailgunTaskBase(JvmToolTaskMixin, TaskBase):
   ID_PREFIX = 'ng'
+  # Possible execution strategies:
+  NAILGUN = 'nailgun'
+  SUBPROCESS = 'subprocess'
 
   @classmethod
   def register_options(cls, register):
     super(NailgunTaskBase, cls).register_options(register)
     register('--use-nailgun', type=bool, default=True,
              help='Use nailgun to make repeated invocations of this task quicker.')
+    register('--execution-strategy', type=str, default='',
```

Comment:
`register` has a `choices` kwarg, so you can:

`register('--execution-strategy', choices=['nailgun', 'subprocess'], ...)`

which will also do the validation for you :)